src/app_service/service/chat/dto/chat.nim

```diff
@@ -120,5 +144,81 @@ proc toChatDto*(jsonObj: JsonNode): ChatDto =
     for memberObj in membersObj:
       result.members.add(toChatMember(memberObj))
 
+  # Add community ID if needed
+  if (result.communityId != "" and not result.id.contains(result.communityId)):
+    result.id = result.communityId & result.id
+
+proc toChannelGroupDto*(jsonObj: JsonNode): ChannelGroupDto =
+  result = ChannelGroupDto()
+  discard jsonObj.getProp("admin", result.admin)
+  discard jsonObj.getProp("verified", result.verified)
+  discard jsonObj.getProp("name", result.name)
+  discard jsonObj.getProp("description", result.description)
+
+  result.channelGroupType = ChannelGroupType.Unknown
+  var channelGroupTypeString: string
+  if (jsonObj.getProp("channelGroupType", channelGroupTypeString)):
+      result.channelGroupType = parseEnum[ChannelGroupType](channelGroupTypeString)
+
+  var chatsObj: JsonNode
+  if(jsonObj.getProp("chats", chatsObj)):
+    for _, chatObj in chatsObj:
+      result.chats.add(toChatDto(chatObj))
+
+  var categoriesObj: JsonNode
+  if(jsonObj.getProp("categories", categoriesObj)):
+    for _, categoryObj in categoriesObj:
+      result.categories.add(toCategory(categoryObj))
+
+  var imagesObj: JsonNode
+  if(jsonObj.getProp("images", imagesObj)):
+    result.images = toImages(imagesObj)
+
+  var permissionObj: JsonNode
+  if(jsonObj.getProp("permissions", permissionObj)):
+    result.permissions = toPermission(permissionObj)
+
+  var membersObj: JsonNode
+  if(jsonObj.getProp("members", membersObj) and membersObj.kind == JObject):
+    for memberId, memberObj in membersObj:
+      result.members.add(toMember(memberObj, memberId))
+
+  discard jsonObj.getProp("canManageUsers", result.canManageUsers)
+  discard jsonObj.getProp("color", result.color)
+  discard jsonObj.getProp("muted", result.muted)
+
+proc mapChatToChatDto*(chat: Chat, communityId: string): ChatDto =
+    result = ChatDto()
+    result.id = chat.id
+    result.communityId = communityId
+    result.name = chat.name
+    result.chatType = ChatType.CommunityChat
+    result.color = chat.color
+    result.emoji = chat.emoji
+    result.description = chat.description
+    result.canPost = chat.canPost
+    result.position = chat.position
+    result.categoryId = chat.categoryId
+    result.communityId = communityId
+
+proc toChannelGroupDto*(communityDto: CommunityDto): ChannelGroupDto =
+  ChannelGroupDto(
+    id: communityDto.id,
+    channelGroupType: ChannelGroupType.Community,
+    name: communityDto.name,
+    images: communityDto.images,
+    chats: communityDto.chats.map(c => c.mapChatToChatDto(communityDto.id)),
```

Comment:
@jrainville in this scope you should use `toChatDto` instead of `mapChatToChatDto`. `mapChatToChatDto` is introduced only because of, now we can say, the old way of handling chats which belongs to community and that is not in place any more. 

Also since `toChannelGroupDto` is called for each "channel group" in case you keep `mapChatToChatDto` here that brings another issue since in line `195`, the field `chatType` is hardcoded to `CommunityChat` which will set chat type for all chats from the "personal" group to `CommunityChat`.